File: 프로그래머스/2/42578. 의상/의상.js
function solution(clothes) {
    let arr = [];
    let cntArr = [];

    for(let i=0; i<clothes.length; i++){
        if(arr.indexOf(clothes[i][1]) === -1){
            arr.push(clothes[i][1]);
            cntArr[arr.indexOf(clothes[i][1])] = 1;
        }
        else cntArr[arr.indexOf(clothes[i][1])]++;
    }
    
    let sum =1;
    for(let i=0; i<cntArr.length; i++){
        sum *= (cntArr[i]+1);
    }
    return sum-1;
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;
unordered_map<string, int> look;

int solution(vector<vector<string>> clothes) {
    for(auto info: clothes) look[info[1]]++;
    
    int sum = 1;
    
    for(auto cloth: look) sum *= cloth.second + 1;
    
    return sum - 1;
}